src/games/strategy/net/OpenFileUtility.java

```diff
@@ -0,0 +1,57 @@
+package games.strategy.net;
+
+import java.awt.Desktop;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+
+import javax.swing.JOptionPane;
+
+import games.strategy.debug.ClientLogger;
+
+/**
+ * A wrapper class for opening Files & URLs using the Desktop API
+ */
+public class OpenFileUtility {
+
+  /**
+   * Opens a specific file on the user's computer using the local computer's file associations.
+   *
+   * @param file The file to be opened
+   */
+  public static void openFile(final File file) {
+    if (Desktop.isDesktopSupported()) {
+      try {
+        Desktop.getDesktop().open(file);
+      } catch (IOException e) {
+        ClientLogger.logError("Could not open File " + file.getAbsolutePath(), e);
+      }
+    } else {
+      logDesktopAPIMessage("File", file.getAbsolutePath());
+    }
+  }
+
+  /**
+   * Opens the specified web page in the user's default browser
+   *
+   * @param url An URL of a web page (ex: "http://www.google.com/")
+   */
+  public static void openURL(final String url) {
+    if (Desktop.isDesktopSupported()) {
+      try {
+        Desktop.getDesktop().browse(URI.create(url));
+      } catch (IOException e) {
+        ClientLogger.logError("Could not open URL " + url, e);
+      }
+    } else {
+      logDesktopAPIMessage("URL", url);
+    }
+  }
+
+  private static void logDesktopAPIMessage(String type, String path) {
+    ClientLogger.logQuietly("The Client doesn't support the Desktop API");
```

Comment:
Still have the double error logging. The point of ClientLogger.logQuietly is to suppress notification from the user,  yet the next line does notification to the user. To boot, this message appearing in the error console is likely to yield any additoinal information.